src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -500,21 +501,56 @@ class WorkspaceService(dataSource: DataSource, workspaceDAO: WorkspaceDAO, entit
     dataSource inTransaction { txn =>
+      txn withGraph { graph =>
       withWorkspace(workspaceName.namespace, workspaceName.name, txn) { workspace =>
         withMethodConfig(workspace, submission.methodConfigurationNamespace, submission.methodConfigurationName, txn) { methodConfig =>
           withEntity(workspace, submission.entityType, submission.entityName, txn) { entity =>
             withMethod(workspace, methodConfig.methodNamespace, methodConfig.methodName, methodConfig.methodVersion, authCookie) { agoraEntity =>
               withWdl(agoraEntity) { wdl =>
-                if (methodConfig.rootEntityType != submission.entityType)
-                  RequestComplete(StatusCodes.Conflict,
-                    s"The method configuration expects an entity of type ${methodConfig.rootEntityType}, but the request describes an entity of type ${submission.entityType}")
-                else MethodConfigResolver.resolveInputsAndAggregateErrors(methodConfig, entity, wdl, txn) match {
-                  case Failure(error) =>
-                    RequestComplete(StatusCodes.BadRequest, "Expression evaluation failed: " + error.getMessage())
-                  case Success(inputsMap) =>
-                    RequestComplete(StatusCodes.Created, executionServiceDAO.submitWorkflow(wdl, MethodConfigResolver.propertiesToWdlInputs(inputsMap), authCookie))
+
+                //If there's an expression, evaluate it to get the list of entities to run this job on.
+                //Otherwise, use the entity given in the submission.
+                val jobEntities: Seq[Entity] = submission.expression match {
+                  case Some(expr) =>
+                    new ExpressionEvaluator(graph, new ExpressionParser()).evalFinalEntity(workspaceName.namespace, workspaceName.name, submission.entityType, submission.entityName, expr) match {
+                      case Success(ents) => ents
+                      case Failure(regret) => return RequestComplete(StatusCodes.BadRequest, "Expression evaluation failed: " + regret.getMessage())
+                    }
+                  case None => List(entity)
+                }
+
+                //Verify the type of all job entities matches the type of the method configuration.
+                if (!jobEntities.forall(_.entityType == methodConfig.rootEntityType)) {
+                  return RequestComplete(StatusCodes.BadRequest, "Entities " + jobEntities.filter(_.entityType != methodConfig.rootEntityType).map(e => e.entityType + ":" + e.name) +
+                    "are not the same type as the method configuration requires")
+                }
+
+                //Attempt to resolve method inputs and submit the workflows to Cromwell, and build the submission status accordingly.
+                val submittedWorkflows = jobEntities.map( e => submitWorkflow(workspaceName, methodConfig, e, wdl, authCookie, txn))
```

Comment:
I'm not sure I understand the question. The failures get collected into `notstarted` and the successes get collected into `workflows`; that information is returned with the submission.